weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/utilities.ts

```diff
@@ -286,6 +286,39 @@ export const objectVersionNiceString = (ov: ObjectVersionSchema) => {
   return result;
 };
 
+export function parseObjectVersionUrlPathToRefUri(
```

Comment:
i'd consider using `usePeekLocation` which does some of this handling for you, and you can just map that to the URI. This will reduce the number of places we are doing manual URI parsing